iceoryx_hoofs/test/moduletests/test_design_newtype.cpp

```diff
@@ -264,12 +259,124 @@ TEST(NewType, AssignByValueMoveComplexTypeDoesCompile)
 TEST(NewType, CreatingNewTypeWithMacroWorks)
 {
     ::testing::Test::RecordProperty("TEST_ID", "d43d41f6-c6d8-4523-a7cf-8f86822643cc");
-    IOX_NEW_TYPE(Sut, uint64_t, newtype::ConstructByValueCopy, newtype::Comparable);
+    IOX_NEW_TYPE(SutType, uint64_t, newtype::ConstructByValueCopy, newtype::Comparable);
 
-    Sut a(73);
-    Sut b(37);
+    SutType a(73);
+    SutType b(37);
     EXPECT_TRUE(a != b);
     EXPECT_FALSE(a == b);
 }
 
+TEST(NewType, NewTypeIsIncrementable)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6d03b24b-fc72-409b-aa2a-f19228ff152c");
+    IOX_NEW_TYPE(SutType,
+                 uint64_t,
+                 newtype::MoveConstructable,
+                 newtype::CopyConstructable,
+                 newtype::Comparable,
+                 newtype::Incrementable,
+                 newtype::ConstructByValueCopy);
+
+    constexpr uint64_t START_VALUE{42};
+    SutType a{START_VALUE};
+    SutType pre{++a};
+    SutType post{a++};
+    EXPECT_THAT(pre, Eq(SutType{START_VALUE + 1}));
+    EXPECT_THAT(post, Eq(SutType{START_VALUE + 1}));
+    EXPECT_THAT(a, Eq(SutType{START_VALUE + 2}));
+}
+
+TEST(NewType, NewTypeIsDecrementable)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "27262c86-2509-4c55-8bff-a37337e79b67");
+    IOX_NEW_TYPE(SutType,
+                 uint64_t,
+                 newtype::MoveConstructable,
+                 newtype::CopyConstructable,
+                 newtype::Comparable,
+                 newtype::Decrementable,
+                 newtype::ConstructByValueCopy);
+
+    constexpr uint64_t START_VALUE{24};
+    SutType a{START_VALUE};
+    SutType pre{--a};
+    SutType post{a--};
+    EXPECT_THAT(pre, Eq(SutType{START_VALUE - 1}));
+    EXPECT_THAT(post, Eq(SutType{START_VALUE - 1}));
```

Comment:
Post decrement returns the old value and then decrements so it should be START_VALUE as well